### PR TITLE
Strategic programmes

### PR DESCRIPTION
**Pull request by davidrapson**
Companion to https://github.com/biglotteryfund/blf-alpha/pull/1141 which adds endpoints for strategic programmes.

Includes a new `sectionBreadcrumbs` field as a step towards getting the content api to return more breadcrumb information. Definitely appreciate thoughts on this.

### CODE REVIEW

File: config/redactor/Standard.json [hunk 1]
```diff
@@ -1,5 +0,0 @@
-{
-  "buttons": ["html", "formatting", "bold", "italic", "unorderedlist", "orderedlist", "link", "image", "video", "file"],
-  "formatting": ["p", "blockquote", "h2", "h3", "h4", "h5"],
-  "plugins": ["fullscreen", "video", "table", "linkButton"]
-}
```
Comment (mattandrews) [inline config/redactor/Standard.json]: This rename might break things in the CMS fields where things are tied to these filenames? eg:

image

File: lib/StrategicProgrammes.php [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            'thumbnail' => $heroImageField ? Images::imgixUrl(
+                $heroImageField->imageMedium->one()->url,
+                ['w' => '640', 'h' => '360']
+            ) : null,
+            'hero' => $heroImageField ? Images::buildHeroImage($heroImageField) : null,
+            'heroCredit' => $entry->heroImageCredit ?? null,
+            'trailText' => $entry->trailText,
+            'intro' => $entry->programmeIntro,
+            'aims' => $entry->strategicProgrammeAims,
+            'aims' => $entry->strategicProgrammeAims,
```
Comment (mattandrews) [inline lib/StrategicProgrammes.php]: Dupe!

File: lib/StrategicProgrammes.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            'impact' => array_map(function ($block) {
+                return [
+                    'title' => $block->contentTitle,
+                    'content' => $block->contentBody,
+                    // 'relatedResearch' => self::extractRelatedResearch($block->relatedResearch)
```
Comment (mattandrews) [inline lib/StrategicProgrammes.php]: Do we need this?

File: lib/StrategicProgrammes.php [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                ];
+            }, $entry->strategicProgrammeImpact->all() ?? []),
+            'aims' => $entry->strategicProgrammeAims,
```
Comment (mattandrews) [inline lib/StrategicProgrammes.php]: Er...

File: config/redactor/Standard.json (hunk 1 above)
Comment (davidrapson) [reply config/redactor/Standard.json]: Tested this out and it defaults to…erm…the 'Default' profile so nothing _breaks_ but I will need to go through each redactor field and re-apply the configs.

File: lib/StrategicProgrammes.php (hunk 3 above)
Comment (davidrapson) [reply lib/StrategicProgrammes.php]: We will _eventually_ but I can remove this for now!

File: lib/StrategicProgrammes.php (hunk 4 above)
Comment (davidrapson) [reply lib/StrategicProgrammes.php]: It's always good to restate your aims

File: config/element-api.php [hunk 5]
```diff
@@ -427,23 +455,25 @@ function getListing($locale)
...
-            if ((string) $entry->outroText) {
-                $entryData['outro'] = $entry->outroText;
-            }
+            $entryData['segments'] = $segments ?? null;
 
-            $segments = parseSegmentMatrix($entry, $locale);
-            if ($segments) {
-                $entryData['segments'] = $segments;
-            }
+            $entryData['outro'] = $entry->outroText ?? null;
```
[the remark this answers is not in the document]
Comment (davidrapson) [reply config/element-api.php]: I'm a big fan of the former, outputting null, makes it clear(er) from any API response what the full schema is.

File: lib/StrategicProgrammes.php (hunk 4 above)
Comment (mattandrews) [reply lib/StrategicProgrammes.php]: `/me golf claps`

File: config/element-api.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (mattandrews) [reply config/element-api.php]: Good call.

**Pull request merged**